### PR TITLE
LPD-21923 Remove FF for FDSViews visualization modes

### PR DESCRIPTION
**Pull request by chestofwonders**
Task: https://liferay.atlassian.net/browse/LPD-21913

**Goal**

The goal is to remove FF LPD-10735 (FDSViews Lists and Cards visualization modes), so by default those visualization modes are added to Tables in FDSViews

**This PR cannot be merged until epic https://liferay.atlassian.net/browse/LPD-10735 was fully completed**

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java [hunk 1]
```diff
@@ -689,33 +688,31 @@ private ObjectDefinition _createFDSViewObjectDefinition(
...
-				defaultVisualizationModeObjectField.getName(),
-				defaultVisualizationModeObjectField.getReadOnly(),
-				defaultVisualizationModeObjectField.
-					getReadOnlyConditionExpression(),
-				defaultVisualizationModeObjectField.isRequired(),
-				defaultVisualizationModeObjectField.isState(),
-				defaultVisualizationModeObjectField.getObjectFieldSettings());
-		}
+		ObjectField defaultVisualizationModeObjectField =
+			ObjectFieldUtil.createObjectField(
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:692]: @chestofwonders Here we need to move lines 692-696 into this [ArrayList](https://github.com/liferay-frontend/liferay-portal/pull/4113/files#diff-39d94b45a2c32f4cf77847688762373395176294708092a3c04cf6f73bfa59f6R634) and delete everything else.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx [hunk 2]
```diff
@@ -54,15 +48,13 @@ let NAVIGATION_BAR_ITEMS = [
...
-		...NAVIGATION_BAR_ITEMS,
-		{
-			Component: Settings,
-			label: Liferay.Language.get('settings'),
-		},
-	];
-}
+NAVIGATION_BAR_ITEMS = [
+	...NAVIGATION_BAR_ITEMS,
+	{
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx:53]: We can move this option directly into the `NAVIGATION_BAR_ITEMS` array

**Comment by juanjofgliferay** [comment]
@chestofwonders @dsanz @markocikos If `Fields` tab is not available anymore, it is worth removing directly `fields.spect.ts` instead of working on fixing the flaky tests. 
After that, it is just a matter of adding the fragment part of the tests into `visualizationModes.spec.ts`

**Comment by markocikos** [comment]
>  If Fields tab is not available anymore, it is worth removing directly fields.spect.ts instead of working on fixing the flaky tests.
After that, it is just a matter of adding the fragment part of the tests into visualizationModes.spec.ts

@juanjofgliferay My idea was to do exactly this in [LPD-20684 Flaky fields tests
](https://liferay.atlassian.net/browse/LPD-20684). I already copied content of `FieldsPage` into `VisualizationModesPage` in https://github.com/liferay-frontend/liferay-portal/pull/4108, so we can delete the page as well.

**Comment by juanjofgliferay** [comment]
@markocikos That was my idea too, but with this PR those tests will be deliverately broken, not flaky. Maybe marking them to be skipped while we work on [LPD-20684 Flaky fields tests.
](https://liferay.atlassian.net/browse/LPD-20684) could be an option.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java (hunk 1 above)
Comment (chestofwonders) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java]: Done!

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx (hunk 2 above)
Comment (chestofwonders) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx:49]: Done!

**Comment by juanjofgliferay** [comment]
@chestofwonders https://github.com/liferay-frontend/liferay-portal/pull/4117 merged
`fields.spec.ts` does not longer exists :sweat:

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by chestofwonders** [comment]
ci:forward

**Comment by markocikos** [comment]
@chestofwonders Did you get the green light from @dsanz and @herrardo to forward this? FF should be removed only when the entire epic is complete.

**Comment by chestofwonders** [comment]
> @chestofwonders Did you get the green light from @dsanz and @herrardo to forward this? FF should be removed only when the entire epic is complete.

Yes, @herrardo gave me the green light

**Comment by chestofwonders** [comment]
ci:stop

**Comment by herrardo** [comment]
go ahead!

**Comment by chestofwonders** [comment]
ci:test:sf

**Comment by chestofwonders** [comment]
ci:forward

**Comment by chestofwonders** [comment]
> @chestofwonders It seems that there is one more LPD-10735 FF in https://github.com/liferay-frontend/liferay-portal/blob/master/portal-impl/src/portal.properties#L6401

I see, thanks @juanjofgliferay!

**Comment by chestofwonders** [comment]
ci:forward

**Comment by chestofwonders** [comment]
ci:forward:force

**Pull request closed without merging**